docs/API.md

```diff
@@ -1038,13 +1038,18 @@ data: {
   head: { key: "value", ... }, // set of string key-value pairs, passed
                                // unchanged from {pub}, optional
   ts: "2015-10-06T18:07:30.038Z", // string, timestamp
-  seq: 123, // integer, server-issued sequential ID
+  seq: 123,   // integer, server-issued sequential ID; optional: could be absent
+              // if message represents a range of deleted messages
+  delseq: [{low: 123, hi: 125}, {low: 156}], // present if message represents a set of ranges
+              // of deleted messages; inclusive-exclusive, i.e. [low, hi), optional
   content: { ... } // object, application-defined content exactly as published
               // by the user in the {pub} message
 }
 ```
 
-Data messages have a `seq` field which holds a sequential numeric ID generated by the server. The IDs are guaranteed to be unique within a topic. IDs start from 1 and sequentially increment with every successful `{pub}`](#pub) message received by the topic.
+Data messages generally have a `seq` field which holds a sequential numeric ID generated by the server. The IDs are guaranteed to be unique within a topic. IDs start from 1 and sequentially increment with every successful [`{pub}`](#pub) message received by the topic.
+
+When user requests a range of messages by issuing a [`{get what=data}`](#get) request, the server responds by sending requested `{data}` messages. If any of the messages in range are deleted, the server also sends one or more `{data}` messages without the `seq` and `content` fields but with `delseq` field. The field contains an array of IDs of deleted message ranges.
```

Comment:
Should we also honor `since`, `before` and `limit` params in `{get what=data}` messages?
It seems logical but OTOH since the purpose of this effort is letting the client figure out the gaps it may be more reasonable to send all the data all the time. I'm personally more inclined to the former. What do you think?

Reply:
> Should we also honor since, before and limit params in {get what=data} messages?

They are honored:
https://github.com/tinode/chat/blob/master/server/db/mysql/adapter.go#L1886